torchao/dtypes/affine_quantized_tensor.py

```diff
@@ -529,6 +558,110 @@ def from_plain(
         return cls(int_data_compressed, scale, zero_point, layout_type)
 
 
+@register_layout_cls(Float8LayoutType)
+class Float8AQTLayout(AQTLayout):
```

Comment:
on mobile, but its unexpected to see this. I thought layout was representing how data is laid out in memory, in which case plain layout sounds right to me.

Reply:
I left a comment about the need for this in the summary:
```
Adds Float8Layout, the main reason for doing this is the ability to pass the mm_config down to matmul via the layout_type. This also adds the ability to do fake "transpose". This may still be needed but I have since learned that we register dispatch entires to nn.fucntional.linear. Thus the weight will not be ran through transpose.
```
@jerryzh168 If this isnt true correct me though please

Reply:
right, I'm saying it seems like this isnt using layout as intended? Do the abstractions need to change instead of tacking something like this on?

Reply:
or, rename it so its not just about layout, etc

Reply:
Is that okay for a follow up PR?

Reply:
layout is an abstraction to track all packing/format related things, this includes the behavior of transpose and how the format behaves in matmul I think